src/script/E2EIdentity/DelayTimer/DelayTimer.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import logdown from 'logdown';
+
+import {util} from '@wireapp/core';
+
+import {getDelayTime} from './delay';
+import {DelayTimerStore} from './DelayTimerStorage';
+
+const {TaskScheduler} = util;
+
+interface CreateGracePeriodTimerParams {
+  gracePeriodInMS: number;
+  gracePeriodExpiredCallback: () => void;
+  delayPeriodExpiredCallback: () => void;
+}
+
+class DelayTimerService {
+  private static instance: DelayTimerService | null = null;
+  private gracePeriodInMS: number;
+  private gracePeriodExpiredCallback: () => void;
+  private delayPeriodExpiredCallback: () => void;
+  private readonly logger = logdown('@wireapp/core/DelayTimer');
+  private delayPeriodTimerKey: string = 'E2EIdentity_DelayTimer';
+  private gracePeriodTimerKey: string = 'E2EIdentity_GracePeriodTimer';
+
+  private constructor({
+    gracePeriodInMS,
+    gracePeriodExpiredCallback,
+    delayPeriodExpiredCallback,
+  }: CreateGracePeriodTimerParams) {
+    this.gracePeriodInMS = gracePeriodInMS;
+    this.gracePeriodExpiredCallback = gracePeriodExpiredCallback;
+    this.delayPeriodExpiredCallback = delayPeriodExpiredCallback;
+    this.initialize();
+  }
+
+  /**
+   * Get the singleton instance of GracePeriodTimer or create a new one
+   * For the first time, params are required to create the instance
+   * After that, params are optional and can be used to update the grace period timer
```

Comment:
Does it really? I have the feeling it does nothing if you call `getInstance` with some new params. Am I missing something?

Reply:
You are right. There is a flaw in my code. Gonna fix that since updating the instance is needed

Reply:
But there is a `update` method, maybe that what is needed (and you can remove that comment)

Reply:
Right. The comment is outdated. Update needed extra functionality (clearing of data) so it has been changed to a method instead